test/jdk/javax/xml/jaxp/datatype/8033980/SerializationTest.java

```diff
@@ -55,122 +52,254 @@
  */
 public class SerializationTest {
 
-    final String FILENAME_CAL = "_XMLGregorianCalendar.ser";
-    final String FILENAME_DURATION = "_Duration.ser";
-    String filePath;
-
-    {
-        filePath = System.getProperty("test.src");
-        if (filePath == null) {
-            //current directory
-            filePath = System.getProperty("user.dir");
-        }
-        filePath += File.separator;
-    }
     final String EXPECTED_CAL = "0001-01-01T00:00:00.0000000-05:00";
     final String EXPECTED_DURATION = "P1Y1M1DT1H1M1S";
     static String[] JDK = {"JDK6", "JDK7", "JDK8", "JDK9"};
 
+    private GregorianCalendarAndDurationSerData[] gregorianCalendarAndDurationSerData = {null, new JDK6GregorianCalendarAndDurationSerData(),
+    new JDK7GregorianCalendarAndDurationSerData(), new JDK8GregorianCalendarAndDurationSerData(), new JDK9GregorianCalendarAndDurationSerData()};
+
     /**
-     * Create the Data files.
+     * Create the Serialization Bytes using JDK under test.
      * @throws DatatypeConfigurationException
      * @throws IOException
+     * @throws ClassNotFoundException
      */
     @BeforeClass
-    public void setup() throws DatatypeConfigurationException, IOException {
+    public void setup() throws DatatypeConfigurationException, IOException, ClassNotFoundException {
         DatatypeFactory dtf = DatatypeFactory.newInstance();
-        for(String jdkVersion : JDK) {
-            if(!Files.exists(Path.of(filePath, jdkVersion+FILENAME_CAL ))) {
-                XMLGregorianCalendar c = dtf.newXMLGregorianCalendar(EXPECTED_CAL);
-                toFile((Serializable) c, filePath + jdkVersion + FILENAME_CAL);
+        XMLGregorianCalendar xmlGregorianCalendar = dtf.newXMLGregorianCalendar(EXPECTED_CAL);
+
+        //Serialize the given xmlGregorianCalendar
+
+        final ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        final ObjectOutputStream oos = new ObjectOutputStream(baos);
+        oos.writeObject(xmlGregorianCalendar);
+        oos.flush();
+        oos.close();
+
+        // Now get a Base64 string representation of the serialized bytes.
+        final String base64 = Base64.getEncoder().encodeToString(baos.toByteArray());
 
+        // Generates the Java Pseudo code for base64 encoded string that can be cut & pasted into the test
+        final StringBuilder sb = new StringBuilder();
```

Comment:
Can you please explain the purpose of the following code in the `setup` method?.  This appears to be what you used to generate the _JDK**X**GregorianCalendarAndDurationSerData.java_ files.  I don't see why this would be needed in a setup method. 

Providing a method, along with comments of how to use it, to create byte arrays make sense, just not in a setup method.

Reply:
This is added in setup method to create the serialized data with current JDK version (JDK under test). Apart from ser data generated with old JDK version.

Reply:
Sorry, I still disagree with this.  Your comment in the test is as follows:

`// Generates the Java Pseudo code for base64 encoded string that can be cut & pasted into the test`

Which is what the method does and to facilitate it, as an example print it out so that it can be copied and pasted

` System.out.println(sb);`

Again, there is no reason to include this code in the setup method, it should be segmented out into its own method with instructions with how to create the encoded string

Reply:
Hi Lance, Initially I misunderstood the comment. Now I move the pseudo code generation code to separate methods.